File: dam_android/jocmemory/app/src/main/java/com/icrescenti/jocmemory/MainActivity.java
package com.icrescenti.jocmemory;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
public class MainActivity extends AppCompatActivity {
    Button play, sortir;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        play = findViewById(R.id.botoMainJugar);
        sortir = findViewById(R.id.botoMainSortir);
        play.setOnClickListener(new View.OnClickListener(){
            public void onClick(View v){
                System.out.println("Iniciant el joc...");
                iniciarJoc();
            }
        });
        sortir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish(); //Tanca l'aplicació - fil principal
            }
        });
    }

    private void iniciarJoc(){
        Intent i = new Intent(this, Joc.class);
        startActivity(i);
    }
}

File: dam_android/CicleVidaNomIvan/app/src/main/java/com/icrescenti/ciclevidanomivan/MainActivity.kt
package com.icrescenti.ciclevidanomivan

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.i("CicleVida", "Executem el mètode 'onCreate'")
    }

    override fun onStart() {
        super.onStart()
        Log.i("CicleVida", "Executem el mètode 'onStart'")
    }

    override fun onResume() {
        super.onResume()
        Log.i("CicleVida", "Executem el mètode 'onResume'")
    }

    override fun onPause() {
        super.onPause()
        Log.i("CicleVida", "Executem el mètode 'onPause'")
    }

    override fun onStop() {
        super.onStop()
        Log.i("CicleVida", "Executem el mètode 'onStop'")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i("CicleVida", "Executem el mètode 'onDestroy'")
    }
}